lib/loader/loader.js

```diff
@@ -110,12 +112,45 @@ var loader, define, requireModule, require, requirejs;
 
   }
 
+  if (DEBUG) {
+    var deprecationLogger = function () {
+      console.warn.apply(console, arguments);
+    };
+
+    var constructMakeDefaultExportDeprecation = function (id) {
+      return 'The `' + id + '` module does not define a default export, but loader.js ' +
+        'generated one anyway. This behavior is deprecated and will be removed in v5.0.0.';
+    };
+
+    Object.defineProperty(requirejs, 'deprecationLogger', {
```

Comment:
I'm not entirely sure what you're getting at, but I simplified things a bit and changed the arguments to support setting `loader.deprecationLogger = Ember.deprecate` if we wanted. What do you think?